test/unit/transformRemappings.spec.ts

```diff
@@ -5,35 +5,137 @@ import mock from 'mock-fs';
+                                  import {FixedPointMathLib as FPL} from 'solmate/utils/FixedPointMathLib.sol';
+                                  import * as FPL from "solmate/utils/FixedPointMathLib.sol";
+                                `;
+
+      const transformedContent = transformRemappings(fileContent);
+
+      expect(transformedContent).to.include(`import {FixedPointMathLib} from 'solmate/src/utils/FixedPointMathLib.sol';`);
+      expect(transformedContent).to.include(`import 'solmate/src/utils/FixedPointMathLib.sol';`);
+      expect(transformedContent).to.include(`import "solmate/src/utils/FixedPointMathLib.sol";`);
+      expect(transformedContent).to.include(`import {FixedPointMathLib as FPL} from 'solmate/src/utils/FixedPointMathLib.sol';`);
+      expect(transformedContent).to.include(`import * as FPL from "solmate/src/utils/FixedPointMathLib.sol";`);
+    });
+  });
 
-  it('should transform imports with remappings', () => {
+  it('should leave unrelated imports alone', () => {
     const fileContent = `
-        import '../../../interfaces/ITest.sol';
-        import './someFile.sol';
-      `;
+      import './Contract.sol';
+      import '../contracts/Contract.sol';
+      import {Contract} from "../contracts/Contract.sol";
+      import {Contract as MyName} from'../contracts/Contract.sol';
+      import * as MyName from'../contracts/Contract.sol';
+    `;
 
     const transformedContent = transformRemappings(fileContent);
 
-    expect(transformedContent).to.include(`import '../../../../../interfaces/ITest.sol';`);
-    expect(transformedContent).to.include(`import './someFile.sol';`);
+    expect(transformedContent).to.include(`import './Contract.sol';`);
+    expect(transformedContent).to.include(`import '../contracts/Contract.sol';`);
+    expect(transformedContent).to.include(`import {Contract} from "../contracts/Contract.sol";`);
+    expect(transformedContent).to.include(`import {Contract as MyName} from'../contracts/Contract.sol';`);
+    expect(transformedContent).to.include(`import * as MyName from'../contracts/Contract.sol';`);
   });
 
-  it('should handle imports from node_modules correctly', () => {
+  it('should leave unrelated statements alone', () => {
     const fileContent = `
-      import '../../../node_modules/some-package/Contract.sol';
-      import 'node_modules/another-package/Token.sol';
+      // import './Contract.sol';
+      // important comment
+      contract C {}
     `;
 
     const transformedContent = transformRemappings(fileContent);
 
-    expect(transformedContent).to.include(`import 'some-package/Contract.sol';`);
-    expect(transformedContent).to.include(`import 'another-package/Token.sol';`);
+    expect(transformedContent).to.include(`// import './Contract.sol';`);
+    expect(transformedContent).to.include(`// important comment`);
+    expect(transformedContent).to.include(`contract C {}`);
+  });
+
+  it.skip('should not have false positives choosing imports', () => {
```

Comment:
why is this test being skipped? we are not able to know if we are talking about code or comment?

Reply:
we are not, since we don't reuse an existing lexer+parser, and only do regular-style modifications to the code. For most cases it is sufficient though, and single-line comments are skipped since we match `\s*import` to detect imports

Reply:
@0xteddybear let's leave a comment to explain why the tests are skipped? I'd also add a warning on the readme.

Reply:
Added the warning in 26e3750 , however I feel leaving the readme as-is and pinning the issue might be better, wdyt?

Reply:
@0xteddybear both are fine in my view